src/java.base/share/classes/jdk/internal/classfile/impl/BufWriterImpl.java

```diff
@@ -152,6 +157,55 @@ public void writeBytes(BufWriterImpl other) {
         writeBytes(other.elems, 0, other.offset);
     }
 
+    @SuppressWarnings("deprecation")
+    void writeUTF(String str) {
+        int strlen = str.length();
+
+        int countGreaterThanZero = 0;
+        byte coder = JLA.stringCoder(str);
+        int freeBytes = strlen + 2;
+        if (coder == 0) {
+            // If it is too long, it may be slow due to cache misses.
+            if (strlen < 256) {
+                countGreaterThanZero = JLA.countGreaterThanZero(str);
+            }
+            freeBytes += strlen - countGreaterThanZero; // 2 bytes
+        } else {
+            freeBytes += (strlen << 1); // 3 bytes
+        }
+        reserveSpace(freeBytes);
+
+        int start = this.offset;
+        int offset = start + 2;
+        byte[] elems = this.elems;
+
+        str.getBytes(0, countGreaterThanZero, elems, offset);
```

Comment:
Might be beneficial to non-latin1 case to move this up into the latin1 block.

Could you verify the 256 limit is useful? Add a couple of cases to the micro with much longer latin1 strings; both ASCII and with non-ASCII at the end. Would be nice if we can keep this simple and the "too long" check feels a bit premature.

Reply:
I verified locally that the 256 limit didn't help, so I removed it. In fact, some MethodTypeDesc may exceed 256.